recmetrics/metrics.py

```diff
@@ -139,6 +139,37 @@ def _ark(actual: list, predicted: list, k=10) -> int:
 
     return score / len(actual)
 
+def _apk(actual: list, predicted: list, k=10) -> int:
+    """
+    Computes the average precision at k.
+    Parameters
+    ----------
+    actual : list
+        A list of actual items to be predicted
+    predicted : list
+        An ordered list of predicted items
+    k : int, default = 10
+        Number of predictions to consider
+    Returns:
+    -------
+    score : int
+        The average precision at k.
+    """
+
+    if len(predicted)>k:
+        predicted = predicted[:k]
+
+    if not predicted:
+        return 0.0
+
+    ## Precision is the number of true positives
+    ##  i.e. the number of things we predict that are
+    ##       actually in `actual`
+    ##  divided by the number of things we predicted
+    precision = sum([True if item in actual else False for item in set(predicted)]) / len(predicted)
```

Comment:
Is this too different of an implementation from `_apk`? If yes, I can do something more like that, but this seemed like quite a simple way to do this. Things we get here:
1. It's just one list comp instead of using `enumerate` and stuff, which IMO makes the math easier to follow
2. Using `for item in set(predicted)` lets us deduplicate the predictions so you don't get double credit for making the same (correct) prediction multiple times.

Question: Is it technically correct to divide by the number of predictions here? I could imagine also dividing by the number of unique predictions. But I think ultimately that boils down to exactly how the metric should be defined